x-pack/plugin/inference/src/main/java/org/elasticsearch/xpack/inference/mapper/SemanticTextFieldMapper.java

```diff
@@ -9,46 +9,98 @@
+import org.elasticsearch.index.mapper.TextFieldMapper;
 import org.elasticsearch.index.mapper.TextSearchInfo;
 import org.elasticsearch.index.mapper.ValueFetcher;
+import org.elasticsearch.index.mapper.vectors.DenseVectorFieldMapper;
+import org.elasticsearch.index.mapper.vectors.SparseVectorFieldMapper;
 import org.elasticsearch.index.query.SearchExecutionContext;
+import org.elasticsearch.inference.SimilarityMeasure;
+import org.elasticsearch.logging.LogManager;
+import org.elasticsearch.logging.Logger;
+import org.elasticsearch.xcontent.XContentBuilder;
 
 import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Iterator;
+import java.util.List;
 import java.util.Map;
+import java.util.Objects;
+
+import static org.elasticsearch.xpack.inference.mapper.InferenceMetadataFieldMapper.INFERENCE_CHUNKS_RESULTS;
+import static org.elasticsearch.xpack.inference.mapper.InferenceMetadataFieldMapper.INFERENCE_CHUNKS_TEXT;
+import static org.elasticsearch.xpack.inference.mapper.InferenceMetadataFieldMapper.RESULTS;
 
 /**
- *  A {@link FieldMapper} for semantic text fields. These fields have a model id reference, that is used for performing inference
- * at ingestion and query time.
- * For now, it is compatible with text expansion models only, but will be extended to support dense vector models as well.
+ * A {@link FieldMapper} for semantic text fields.
+ * These fields have a reference id reference, that is used for performing inference at ingestion and query time.
  * This field mapper performs no indexing, as inference results will be included as a different field in the document source, and will
- * be indexed using {@link InferenceResultFieldMapper}.
+ * be indexed using {@link InferenceMetadataFieldMapper}.
  */
 public class SemanticTextFieldMapper extends FieldMapper {
+    private static final Logger logger = LogManager.getLogger(SemanticTextFieldMapper.class);
 
     public static final String CONTENT_TYPE = "semantic_text";
 
     private static SemanticTextFieldMapper toType(FieldMapper in) {
         return (SemanticTextFieldMapper) in;
     }
 
-    public static final TypeParser PARSER = new TypeParser((n, c) -> new Builder(n), notInMultiFields(CONTENT_TYPE));
+    public static final TypeParser PARSER = new TypeParser(
+        (n, c) -> new Builder(n, c.indexVersionCreated(), c.getIndexAnalyzers()),
+        notInMultiFields(CONTENT_TYPE)
+    );
+
+    private final IndexVersion indexVersionCreated;
+    private final SemanticTextModelSettings modelSettings;
+    private final IndexAnalyzers indexAnalyzers;
+    private final NestedObjectMapper subMappers;
 
-    private SemanticTextFieldMapper(String simpleName, MappedFieldType mappedFieldType, CopyTo copyTo) {
+    private SemanticTextFieldMapper(
+        String simpleName,
+        MappedFieldType mappedFieldType,
+        CopyTo copyTo,
+        IndexVersion indexVersionCreated,
+        IndexAnalyzers indexAnalyzers,
```

Comment:
It seems a bit silly that we're defining the index analyzers, given that we're not indexing or storing the text field.

The point of this text field is to store the text that generated the chunk in `_source`, correct? What if we made it a `keyword` field instead?

Reply:
Yep good call, I wonder if we need to explicitly map this field though. Maybe we can just avoid creating it in the mapping entirely?

Reply:
@carlosdelest WDYT?

Reply:
I don't think we want to index or analyze it in any way; users can use multifields / copy_to in case they want to search for it. I'd vote for skipping the mapping.

Reply:
I switched to a keyword field instead so that it's clear that the field is there but not indexed.